datafusion/optimizer/src/replace_distinct_aggregate.rs

```diff
@@ -94,10 +96,18 @@ impl OptimizerRule for ReplaceDistinctWithAggregate {
             })) => {
                 let expr_cnt = on_expr.len();
 
+                let first_value_udaf = func_registry.udaf("first_value")?;
                 // Construct the aggregation expression to be used to fetch the selected expressions.
-                let aggr_expr = select_expr
-                    .into_iter()
-                    .map(|e| first_value(vec![e], false, None, sort_expr.clone(), None));
+                let aggr_expr = select_expr.into_iter().map(|e| {
```

Comment:
this change is the reason for this PR I think -- to avoid the hard coded dependency on `first_value`...